src/framework/MOM_io_file.F90

```diff
@@ -1729,35 +1729,36 @@ subroutine get_field_nc(handle, label, values, rescale)
 
   field_nc = handle%fields%get(label)
 
-  values(:,:) = 0.0
-  if (data_domain) then
-    allocate(values_nc(isc:iec,jsc:jec), source=0.)
-    iscl = isc+1-isd ; iecl = iec+1-isd ; jscl = jsc+1-jsd ; jecl = jec+1-jsd
-  else
-    iscl = 1 ; iecl = iec+1-isc ; jscl = 1 ; jecl = jec+1-jsc
```

Comment:
Even in the case when `data_domain = false`, `iscl`, `iecl`, `jscl` and `jecl` are used for the array extents when rescaling the field on line ~1764 when `rescale /= 1`, but in this suggested revision to the PR, they are undefined in this case.  So this modification to the PR changes a great many things, but unfortunately it reintroduces a new segmentation fault on line 1764, exactly where one was before that was causing the runs to fail. This can be verified by setting `Z_RESCALE_POWER = 2` and `T_RESCALE_POWER = 3` (or any other values than 0) in the global_ALE test cases and many others. The previous version of the PR does fix the bug it was intended to fix; this modified version does not.

  Please revise these modifications to address the problem that necessitated this PR in the first place, as documented at https://github.com/NOAA-GFDL/MOM6/issues/333.